### DISCUSSION TITLE
Segmenting EBSD map

### DISCUSSION THREAD

**Discussion by hannahcunning** [mtex-toolbox/mtex#1834] [Ask Anything] [answered]
Hello, 

I am trying to segment my ebsd map into two areas, one that contains the dolomite clast and the other that has the beginning of the clast tail so that I can look at differences in the pole figures etc. The fact that I am wanting to make the segments non-rectangles is giving me some trouble. 
`%Clast area
pgon=polyshape([0 169 138 85 0],[0 0 129 200 200]);`
![area](https://github.com/mtex-toolbox/mtex/assets/54119012/12ff1b58-4cf1-431d-b4ed-181a4fd1ef61)

I tried doing  
`condition = inpolygon(ebsd,region); 
ebsdclast = ebsd(condition);` 
but since the area I am wanting is not a rectangle it doesn't work. 
`Index in position 2 exceeds array bounds. Index must not exceed 1.
Error in EBSD/inpolygon (line 34)
  ind = insidepoly(ebsd.prop.x,ebsd.prop.y,xy(:,1),xy(:,2));`

I thought about dividing the areas into some sizeable rectangles and removing grains from the rectangles and then removing individual grains , but I don't really want to do `ebsdClast(grainsClast(##))=[];` forever. 

Is there an easier way to do this?

Thanks, Hannah

**Comment by kilir** [comment, marked as answer]
Hi Hannah,
just make sure your `region` is n-by-2 list of x and y coordiantes describing yout polygon you want to extract.

```
plot(ebsd)
xy = ginput() % clickidiclick your polygon, hit return/enter when done
ebsd_sub = ebsd(inpolygon(ebsd,xy))
```

Hope that helps.

Cheers,
Rüdiger

**Reply by hannahcunning** [reply]
Thanks so much!

**Reply by AzdiarGazder** [reply]
Hi @hannahcunning,

To help you segment your map, please refer to the "crop" function in the following weblink: https://github.com/AzdiarGazder/mtexTools.

Warm regards,
Azdi
